src/scheduling_algorithms.c

```diff
@@ -0,0 +1,90 @@
+#include "headers.h"
+
+//! syntax modification when merging with others for process variable data members
+
+/**
+ * updateCurrentProcess - update the current process data
+ *
+ * @param head: Pointer to the pointer to the head of the priority queue.
+ * @param current_process: Pointer to the current running process by the cpu.
+ * just for code simplification.
+ */
+
+void updateCurrentProcess(pqueue_t **head, rprocess_t *current_process)
+{
+    if (current_process == NULL)
+        current_process = (rprocess_t *)(sizeof(rprocess_t));
+    current_process->process = *head->process;
+    current_process->priority = *head->priority;
+    pop(head);
```

Comment:
`pop()` frees the node from the memory so the current_process will point to garbage